### DISCUSSION TITLE
Is there a way to specify the file extension based on URL when opening the external editor?

### DISCUSSION THREAD

**Discussion by xuhdev** [tridactyl/tridactyl#5127] [Q&A]
It looks like tridactyl always saves the temporary file with a `.txt` suffix.

I would like to let my text editor recognize that the type of the file
based on the suffix, which I would set based on URL.

For example, I would set `.md` as the suffix for all text editing on github.com.

**Comment by jman-schief** [comment]
I am not sure I understand the question but I think what you're asking is best accomplished outside Tridactyl. I would look into how your operating system or window manager handles opening files applications based on their extension.

**Reply by xuhdev** [reply]
Sorry for being unclear. I meant to have a way to decide what file extension Tridactyl saves as. For example, if it's `github.com`, I would like Tridactyl to save all files as `.md` files. On most websites, I would like to edit as `.txt` files.

**Reply by bovine3dom** [reply]
Can't you use an autocmd in Vim that sets the file type based on the filename, which includes the domain?

**Reply by xuhdev** [reply]
@bovine3dom That's a good way to solve it! But there seems to be no way to tell if the file is from Tridactyl. Perhaps adding a `tridactyl` prefix to all files can solve this problem?

**Reply by bovine3dom** [reply]
What's wrong with matching `tmp_[domain]_[alphanumeric].txt` ?

I can't imagine many other programs are making files like that
